pxr/usd/sdf/childrenView.h

```diff
@@ -76,27 +75,75 @@ template <typename _Owner, typename _InnerIterator, typename _DummyPredicate>
 class Sdf_ChildrenViewTraits {
 private:
 
-    // Internal predicate object which will be passed to the filter
-    // iterator. This just calls through to the owner's predicate.
-    class _Predicate {
+    // Owner's predicate object will be used by the filter iterator.
+    // In C++20, consider using the ranges library to simplify this
+    class _FilterIterator {
     public:
-        typedef typename _Owner::value_type value_type;
+        using iterator_category = std::forward_iterator_tag;
+        using value_type = typename _InnerIterator::value_type;
+        using reference = typename _InnerIterator::reference;
+        using pointer = typename _InnerIterator::pointer;
+        using difference_type = typename _InnerIterator::difference_type;
+
+        _FilterIterator() = default;
+        _FilterIterator(const _Owner* owner,
+                        const _InnerIterator& underlyingIterator,
+                        const _InnerIterator& end) :
+                            _owner(owner),
+                            _underlyingIterator(underlyingIterator),
+                            _end(end) { filter(); }
+
+        reference operator*() const { return *_underlyingIterator; }
+        pointer operator->() const { return _underlyingIterator.operator->(); }
+
+        _FilterIterator& operator++() {
+            TF_DEV_AXIOM(_underlyingIterator != _end);
+            ++_underlyingIterator;
+            filter();
+            return *this;
+        }
+
+        _FilterIterator operator++(int) {
+            TF_DEV_AXIOM(_underlyingIterator != _end);
+            _FilterIterator result(*this);
+            ++_underlyingIterator;
+            filter();
+            return result;
+        }
+
+        bool operator==(const _FilterIterator& other) const {
+            return _underlyingIterator == other._underlyingIterator;
+        }
+
+        bool operator!=(const _FilterIterator& other) const {
+            return _underlyingIterator != other._underlyingIterator;
+        }
 
-        _Predicate() : _owner(NULL) { }
-        _Predicate(const _Owner* owner) : _owner(owner) { }
+        const _InnerIterator& GetBase() const { return _underlyingIterator; }
 
-        bool operator()(const value_type& x) const
+    private:
+        // Skip any iterators that don't satisfy the predicate
```

Comment:
These `shouldFilter` and `filter` functions aren't here for API conformance right? If I'm right, they should follow our coding conventions for private methods, i.e. capitalized and starting with an underscore